compiler/src/dotty/tools/dotc/inlines/Inliner.scala

```diff
@@ -1062,15 +1062,23 @@ class Inliner(val call: tpd.Tree)(using Context):
    *  This corresponds to the symbols that will need to be interpreted.
    */
   private def macroDependencies(tree: Tree)(using Context) =
+    val typeAccumulator = new TypeAccumulator[List[Symbol]] {
```

Comment:
It's hard to figure out if this is good enough because it's hard to know which trees are valid here. I think it'd be much clearer if we combined this logic and the checks from checkIfValidStaticCall in the same traversal, so we'd know exactly which tree shape we need to handle.

By the way, checkIfValidStaticCall is flawed because of local imports:
```scala
object Inline extends Macro {
  val x: Macro = ???
  import x._
  inline def callMacro(): Int =
    ${ impl() }
}
```